presto-main/src/main/java/com/facebook/presto/server/TaskResource.java

```diff
@@ -319,6 +324,20 @@ private static boolean shouldSummarize(UriInfo uriInfo)
         return uriInfo.getQueryParameters().containsKey("summarize");
     }
 
+    private static <T> ListenableFuture<T> addTimeout(ListenableFuture<T> future, Callable<T> onTimeout, Duration timeout, ScheduledExecutorService executorService)
```

Comment:
This should go as a separate commit as a util method in a future-related util class